Integrate new reporting ui

kldavis4 opened the pull request.

edwinsteele commented:
Looking through the reporting UI PR. I’m getting “Unexpected Error!” when viewing both the “Top 10" and “All Stats”. I’ve had the machine running for a day, but haven’t browsed to any content until a few minutes before attempting to view the stats. At the same time I’m seeing the following in the logs, which might be linked (though I thought we created an empty file for each of these… am I wrong?)

```
2017/12/24 21:04:22 [error] 18599#18599: *2 open() "/var/www/connectbox/connectbox_default/stats.top10.json" failed (2: No such file or directory), client: 172.28.128.1, server: stretch-vagrant.connectbox, request: "GET /stats.top10.json HTTP/1.1", host: "stretch-vagrant.connectbox", referrer: "http://stretch-vagrant.connectbox/admin/reports/index.html?report=stats.top10"
2017/12/24 21:04:29 [error] 18599#18599: *2 open() "/var/www/connectbox/connectbox_default/stats.json" failed (2: No such file or directory), client: 172.28.128.1, server: stretch-vagrant.connectbox, request: "GET /stats.json HTTP/1.1", host: "stretch-vagrant.connectbox", referrer: "http://stretch-vagrant.connectbox/admin/reports/index.html?report=stats”
```

edwinsteele commented:
File names aren’t appearing properly in the reporting UI (I went there directly, given the comment above). These are all files from the sample content, so you can reproduce easily. The second in the list is the Arabic wikipedia page - i’m not sure whether it’s being presented with RTL (my on-the-fly unencoding isn’t that good lol)
<img width="989" alt="pervasive-percent" src="https://user-images.githubusercontent.com/1515215/34342717-af05be7a-ea0c-11e7-9d2b-a596b083b376.png">
The

edwinsteele commented on ansible/roles/webserver-content/tasks/main.yml:
```diff
@@ -37,6 +37,21 @@
   args:
     chdir: /tmp/connectbox_client/{{ connectbox_client_path }}
 
+- name: retrieve connectbox-reports-repo
```
I realise this is the pattern we’ve used in the past, but I can’t remember why we’re doing it as opposed to creating the directory and cloning directly into that new directory. I’d prefer to clone into the new directory - can you make that change if you can’t think of a good reason to keep it like this? (I’m happy to apply the pattern to our other clones as a part of a cleanup)

edwinsteele commented:
When viewing the top 10 reports, there’s no need to show (or allow selection of) the drop-down that allows 20 or 100 rows. I thought something was wrong with the UI, as I didn’t notice I was in the top 10 and knew I had more than 10 items.

edwinsteele commented:
Once I follow a reporting link, the only way back to the admin UI appears to be URL hacking. Can we put the reporting UI inside the admin interface, or at least have a return link?

edwinsteele commented:
Also noticing that reports do not show updated data without an explicit browser reload e.g.

1. view report for current week
2. look at some content
3. wait for the reporting cron job to run
4. switch to view another week report via the dropdown
5. switch to view report for current week via the dropdown

At step 5, I expect to see the data that reflects the views in step 2 (that was processed in step 3), but the new data isn’t available. I suspect we need an expiry directive so to avoid caching the stats json responses (we have something similar in the icon-only nginx config to avoid caching index responses, if you want an example)

edwinsteele commented:
Also, changing the page via the up and down arrows doesn’t actually change the page (changing the page number directly and pressing enter does, as does the prev and next link)

kldavis4 commented:
I _think_ I addressed all of the PR feedback for this except for the ansible git checkout thing. The issue is that I need to checkout a subdirectory of the react project (/published) and I don't know of any way to do that using the ansible git command. So it checks out the entire repo and copies that directory.

edwinsteele commented:
Fair enough re: the git thing. I know it’s possible to do a command line npm install from a GitHub branch. Perhaps we can pursue that path later using the npm ansible task if the reporting UI is a node package. Most things look good. 2 things that still aren’t quite right:

* In the top-10 requests, the period dropdown has day, week, month, hour, year instead of hour, day, week, month, year (it’s possible that I missed this in my first review - it’s not necessarily a regression with this last round of changes)
* I’m still seeing stale stats with the 5-step process that I describe above, but I do notice that the stats are updated if I click the left arrow (to go back to the admin interface) and then select the same report.

kldavis4 commented:
I've added sorting to the filters (it was just taking the order from the report json) and I've modified it so it reloads the report json whenever the view filters are changed which should address the issues you reported.

edwinsteele commented:
Thank you. Wasn’t able to re-test (I’m in the middle of a computer rebuild) but feel free to merge.
